### ISSUE TITLE
Input and output variables

### ISSUE THREAD

**Issue by jkweon** [MahmoudAbdelRahman/GH_CPython#2]
Thanks for this valuable development.

Please note the error shown on the following image.

Can the component accept input? How can I use input part with text, number, geometry and so on? Please note the error message on the python test window.

![20170719 001](https://user-images.githubusercontent.com/859448/28355909-f7ac5592-6ca0-11e7-88d4-05a6adb1ea39.jpg)

I also found output variable did not work. The panel should show 100 if the component works.

![20170719 002](https://user-images.githubusercontent.com/859448/28355947-1bfcce9a-6ca1-11e7-8ffd-3501d9e16582.jpg)

Would you show me how to handle input and output variables? If input and output variable parts are under development, a couple of basic types can be used in stead of  all types of objects until the development is completed. 

Thanks in advance.

Jihoon

**Comment by MahmoudAbdelRahman** [comment]
Hi Jihoon, 
Thanks so much for your feedback, 
I have been working on this problem all the day, finally I've come up with some solutions. Although it requires more work to get rid of some bugs, **I have connected the inputs with the script**. 
![workingonproblem](https://user-images.githubusercontent.com/6969514/28442262-cadcc67c-6daf-11e7-9f1e-896e2e3ebd24.JPG)
I'm now working on the outputs. 
Sincere regards, 
Mahmoud Abdelrahman.

**Comment by jkweon** [comment]
That is a good improvement. Thank you, Mahmoud. I am looking foward to your output improvement. 

I have one more suggestion. Is it possible for the componemt to import rhincommon libraries?  Python for. NET might be one way for achieving it, but I am not sure. I don't know if you already done this. It will be a way to go for the use of all existing resources from Rhino.

Thank you very much.

Jihoon

**Comment by MahmoudAbdelRahman** [comment]
@jkweon 
Hi Jihoon, 
Thanks so much I have added some improvements to the inputs, I'm working now on the output as well as saving data. I'll try to handle RhinoCommon or .Net as possible, I think there is a way to do this. I'll give it a try. 
Regarding Colors, yes, I've just noticed that copying GHPython code changes colors, I'm going to improve the whole highlight and autocomplete system using ICSharpCode.TextEditor. 
Thanks so much. 
Mahmoud Abdlerahman
